### PR TITLE
Kafka Trigger Rich Type

### PR DESCRIPTION
**Pull request by priyaananthasankar**
Supporting rich type KafkaEvent for KafkaTrigger in Python
Azure/azure-functions-kafka-extension#10

### CODE REVIEW

File: azure/functions_worker/bindings/kafka.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from . import meta
+from .. import protos
+
+
+class KafkaConverter(meta.InConverter, meta.OutConverter,
+                        binding='kafka'):
+
+    @classmethod
+    def check_input_type_annotation(
+           cls, pytype: type, datatype: protos.BindingInfo.DataType) -> bool:
```
Comment (priyaananthasankar) [inline azure/functions_worker/bindings/kafka.py]: Added extra argument - in sync with meta.py

File: azure/functions_worker/bindings/kafka.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            body = data.json.encode('utf-8')
+
+        else:
+            raise NotImplementedError(
+                f'unsupported event data payload type: {data_type}')
+
+        return _kafka.KafkaEvent(
+            body=body,
+            timestamp=cls._decode_trigger_metadata_field(
+                trigger_metadata, 'Timestamp', python_type=str),
```
Comment (maiqbal11) [inline azure/functions_worker/bindings/kafka.py]: Mentioned in https://github.com/Azure/azure-functions-python-library/pull/13 as well. Keep this as `datetime.datetime`?

Comment (priyaananthasankar) [reply azure/functions_worker/bindings/kafka.py]: Given the different kinds of datetime formats that arrive and there is no specification on what should be a time format, I thought its best to leave it as string and allow folks to convert the way they want it.

**Comment by maiqbal11** [comment]
@priyaananthasankar, looks good to me overall. Once https://github.com/Azure/azure-functions-python-worker/pull/398 is checked in and released, would be better to check this change into the Kafka extensions repo. You can include the entry point into the `setup.py` for the kafka trigger package. This would be a cleaner way of just collecting all the kafka extension related code in one place (this is a more general pattern we are recommending). \cc @pragnagopa

**Comment by pragnagopa** [comment]
@maiqbal11 - Can we get a release for the python worker?
@priyaananthasankar  - As @maiqbal11 mentioned please move this PR to  https://github.com/Azure/azure-functions-kafka-extension

**Comment by maiqbal11** [comment]
@pragnagopa, I'll start the release.

**Comment by maiqbal11** [comment]
Implemented in https://github.com/Azure/azure-functions-kafka-extension/pull/92

**Pull request closed without merging**